### PR TITLE
test(unit): add unit test for many components

### PR DESCRIPTION
**Pull request by ej9x**

### CODE REVIEW

File: src/components/AsyncContent/AsyncContent.test.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// @flow
```
Comment (zouxuoz) [inline src/components/AsyncContent/AsyncContent.test.js]: @ej9x why we can't use snapshots to check this component?

File: src/components/Breadcrumbs/Breadcrumbs.test.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+// @flow
```
Comment (zouxuoz) [inline src/components/Breadcrumbs/Breadcrumbs.test.js]: @ej9x why we can't use snapshots to check this component?

File: src/components/Button/Button.js [hunk 3]
```diff
@@ -200,9 +200,9 @@ class Button extends Component<ButtonProps> {
   };
 
   onClick = (event: *) => {
-    const { onClick, disabled } = this.props;
+    const { onClick, disabled, loading } = this.props;
 
-    !disabled && onClick && onClick(event);
+    !disabled && !loading && onClick && onClick(event);
```
Comment (zouxuoz) [inline src/components/Button/Button.js]: @ej9x we already disable the ability to interact with `disabled` / `loading` button via CSS `pointer-events: none`. Can we delete this logic?

File: src/components/Button/Button.test.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
+// @flow
+import React from 'react';
+import { Button } from './Button';
+
+describe('<Button />', () => {
+
+  it('should render button with text', () => {
```
Comment (zouxuoz) [inline src/components/Button/Button.test.js]: @ej9x I think the tests for check render should use snapshots, it is more easy for writing, supporting and reading

File: src/components/CheckboxField/CheckboxField.test.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+// @flow
+import React from 'react';
+import { CheckboxField } from './CheckboxField';
+
+describe('<Checkbox />', () => {
```
Comment (zouxuoz) [inline src/components/CheckboxField/CheckboxField.test.js]: @ej9x Why we test same functionality in `Checkbox` and `CheckboxField`?

File: src/components/DateInput/DateInput.js [hunk 6]
```diff
@@ -20,8 +20,8 @@ injectGlobal`
 type DateInputProps = {|
   onChange: (value: ?string) => void,
   value: ?string,
-  withTime?: boolean,
-  stretch?: boolean,
+  withTime ?: boolean,
```
Comment (zouxuoz) [inline src/components/DateInput/DateInput.js]: @ej9x As I know, we don't use space before `?`

Comment (ej9x) [reply src/components/DateInput/DateInput.js]: fixed

File: src/components/Breadcrumbs/Breadcrumbs.test.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+      { path: '/app/settings', label: 'Settings' },
+    ];
+
+    const wrapper = mount(<Breadcrumbs pathname="/app/settings/security" routes={ routes } param={ 1 } />);
+
+    expect(wrapper.find(Link).at(1).text()).toBe('1');
+    expect(wrapper.find(Link).at(1).prop('to')).toBe('/app');
+  });
+
+  it('should render piece of breadcrumbs with not valid path', () => {
```
[the remark this answers is not in the document]
Comment (ej9x) [reply src/components/Breadcrumbs/Breadcrumbs.test.js]: fixed

File: src/components/CheckboxField/CheckboxField.test.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    Object {
+      "checked": true,
+      "onChange": [MockFunction],
+    }
+  }
+  meta={Object {}}
+  stretch={true}
+>
+  <Checkbox
+    checked={false}
```
[the remark this answers is not in the document]
Comment (ej9x) [reply src/components/CheckboxField/CheckboxField.test.js]: fixed

**Pull request merged**